axelor-stock/src/main/resources/views/StockLocation.xml

```diff
@@ -240,6 +244,12 @@
       method="setStocklocationValue"/>
   </action-method>
 
+  <action-record name="action-stock-location-record-set-stock-location-value-Btn"
+    model="com.axelor.apps.stock.db.StockLocation">
+    <field name="$stockLocationValueBtn"
+      expr="eval: java.text.NumberFormat.getInstance(new Locale(__user__.localization?.code)).format(stockLocationValue)"/>
```

Comment:
the format does not work, please:

- to create the locale, use `Locale.forLanguageTag()` method and pass the `language` field to the method instead of the localization code
- Manage the case where the user language is empty by using Locale.US by default

maybe it would be interesting to create a method in a service annotated with `@CallMethod` to format a bigdecimal